test/test-vcf-api.c

```diff
@@ -527,6 +527,26 @@ void test_invalid_end_tag(void)
     hts_set_log_level(logging);
 }
 
+void test_open_format() {
+    char mode[5];
+    strcpy(mode, "r");
+    vcf_open_mode(mode+1, "mode1.bcf", NULL);
+    if (strncmp(mode, "rb", 2))
+        error("Mode '%s' does not match the expected value '%s'", mode, "rb");
+    mode[1] = 0;
+    vcf_open_mode(mode+1, "mode1.vcf", NULL);
+    if (strncmp(mode, "r", 1))
+        error("Mode '%s' does not match the expected value '%s'", mode, "r");
+    mode[1] = 0;
+    vcf_open_mode(mode+1, "mode1.vcf.gz", NULL);
+    if (strncmp(mode, "rz", 2))
+        error("Mode '%s' does not match the expected value '%s'", mode, "rz");
+    mode[1] = 0;
+    vcf_open_mode(mode+1, "mode1.vcf.bgz", NULL);
+    if (strncmp(mode, "rz", 2))
+        error("Mode '%s' does not match the expected value '%s'", mode, "rz");
+}
```

Comment:
This could do with a test with an invalid (for vcf/bcf) extension to ensure `vcf_open_mode()` returns -1 in such a case.  See comments in `vcf_open_mode()`.